public/app/features/plugins/admin/pages/PluginDetails.test.tsx

```diff
@@ -332,4 +342,40 @@ describe('Plugin details page', () => {
     // Check if the modal disappeared
     expect(queryByText('Uninstall Akumuli')).not.toBeInTheDocument();
   });
+
+  it('should not display the install / uninstall / update buttons if the GCOM api is not available', async () => {
+    let rendered: RenderResult;
+    const plugin = getCatalogPluginMock({ id });
+    const state = getPluginsStateMock([plugin]);
+
+    // Mock the store like if the remote plugins request was rejected
+    const pluginsStateOverride = {
+      ...state,
+      requests: {
+        ...state.requests,
+        [fetchRemotePlugins.typePrefix]: {
+          status: RequestStatus.Rejected,
+        },
+      },
+    };
+
+    // Does not show an Install button
+    rendered = renderPluginDetails({ id }, { pluginsStateOverride });
+    await waitFor(() => expect(rendered.queryByRole('button', { name: /(un)?install/i })).not.toBeInTheDocument());
+    rendered.unmount();
+
+    // Does not show a Uninstall button
+    rendered = renderPluginDetails({ id, isInstalled: true }, { pluginsStateOverride });
+    await waitFor(() => expect(rendered.queryByRole('button', { name: /(un)?install/i })).not.toBeInTheDocument());
+    rendered.unmount();
+
+    // Does not show an Update button
+    rendered = renderPluginDetails({ id, isInstalled: true, hasUpdate: true }, { pluginsStateOverride });
+    await waitFor(() => expect(rendered.queryByRole('button', { name: /update/i })).not.toBeInTheDocument());
+
+    // Shows a message to the user
+    // TODO<Import these texts from a single source of truth instead of having them defined in multiple places>
+    const message = 'The install controls have been disabled because the Grafana server cannot access grafana.com.';
```

Comment:
If you're gonna go with the earlier suggestion this will need updating too.

```suggestion
    const message = 'The functionality to install and uninstall plugins has been disabled because the Grafana server cannot access grafana.com.';
```